nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/action/TrashAction.java

```diff
@@ -120,6 +123,13 @@ public void setSystemProperty(CoreSession session, List<String> ids, Boolean val
                     DocumentRef ref = new IdRef((String) map.get(ECM_UUID));
                     try {
                         session.setDocumentSystemProp(ref, "isTrashed", value);
+                        String docName = (String) map.get(ECM_NAME);
+                        TrashService trashService = Framework.getService(TrashService.class);
```

Comment:
just extract a variable outside the for loop to avoid calling many time the same service